front/src/common/Map/Search/useSearchOperationalPoint.tsx

```diff
@@ -36,46 +44,64 @@ export default function useSearchOperationalPoint({
   const debouncedSearchTerm = useDebounce(searchTerm, debounceDelay);
   const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
 
-  const searchOperationalPoints = async () => {
-    const isSearchingByTrigram =
+  const searchOperationalPointsByTrigram = async (
+    infraId: number,
+    dpyToMasOperationalpointsFilter: SearchQuery
+  ) => {
+    const shouldSearchByTrigram =
       !Number.isInteger(+debouncedSearchTerm) && debouncedSearchTerm.length < 4;
-    const searchQuery = isSearchingByTrigram
-      ? // We have to test for op names that goes under 4 letters too
-        ['or', ['=i', ['trigram'], debouncedSearchTerm], ['search', ['name'], debouncedSearchTerm]]
-      : [
-          'or',
-          ['search', ['name'], debouncedSearchTerm],
-          ['like', ['to_string', ['uic']], `%${debouncedSearchTerm}%`],
-        ];
-
-    const dpyToMasOperationalpointsFilter =
-      isStdcm && !isSuperUser
-        ? [
-            'or',
-            ...DPY_TO_MAS_OPERATIONAL_POINTS.map(([ci, ch]) => [
-              'and',
-              ['=', ['ci'], ci],
-              ['=', ['ch'], ch],
-            ]),
-          ]
-        : true;
+
+    if (!shouldSearchByTrigram) return [];
 
     const payload = {
       object: 'operationalpoint',
       query: [
         'and',
-        searchQuery,
-        infraID !== undefined ? ['=', ['infra_id'], infraID] : true,
+        ['=i', ['trigram'], debouncedSearchTerm],
+        ['=', ['infra_id'], infraId],
         dpyToMasOperationalpointsFilter,
       ],
     };
-
     try {
       const results = (await postSearch({
         searchPayload: payload,
         pageSize: 101,
       }).unwrap()) as SearchResultItemOperationalPoint[];
-      setSearchResults(results);
+      return results;
+    } catch (error) {
+      setFailure(castErrorToFailure(error));
+      return [];
+    }
+  };
+
+  const searchOperationalPoints = async () => {
+    if (infraID === undefined) return;
+
+    const dpyToMasOperationalpointsFilter = isStdcm && !isSuperUser ? DPY_TO_MAS_FILTER : true;
+
+    const trigramResults = await searchOperationalPointsByTrigram(
+      infraID,
+      dpyToMasOperationalpointsFilter
+    );
+
+    try {
+      const results = (await postSearch({
+        searchPayload: {
+          object: 'operationalpoint',
+          query: [
+            'and',
+            [
+              'or',
+              ['search', ['name'], debouncedSearchTerm],
+              ['like', ['to_string', ['uic']], `%${debouncedSearchTerm}%`],
+            ],
+            ['=', ['infra_id'], infraID],
+            dpyToMasOperationalpointsFilter,
+          ],
+        },
+        pageSize: 101,
+      }).unwrap()) as SearchResultItemOperationalPoint[];
+      setSearchResults([...trigramResults, ...results]);
```

Comment:
Do we need to de-duplicate results here? e.g. if both search queries return the same OP, it would appear twice in the list?

Reply:
We have a post-processing functions that filter the duplicates :) So I think it's ok like this